messenger/decoder.go

```diff
@@ -139,22 +139,36 @@ func (d *httpDecoder) run(res http.ResponseWriter) {
 		log.V(2).Infoln(d.idtag + "run: terminating")
 	}()
 
+	connCh := make(chan net.Conn, 1)
 	go func() {
+		con, ok := <-connCh
+		if !ok {
+			log.V(3).Infoln(d.idtag + "run: response-writer aborted")
+			return
+		}
+		defer func() {
+			log.V(3).Infoln(d.idtag + "run: closing connection")
+			con.Close()
+		}()
 		for buf := range d.outCh {
-			select {
-			case <-d.forceQuit:
-				return
-			default:
-			}
 			//TODO(jdef) I worry about this busy-looping
-			for buf.Len() > 0 {
+			now := time.Now()
+			for buf.Len() > 0 && time.Since(now) < 30*time.Second {
```

Comment:
the logic in this double for loop and select statement is a bit complex. Can you leave some comments about what you're doing to make it easier to read?